builtin/v12/migration/miner.go

```diff
@@ -171,11 +213,18 @@ func migrateSectorsWithCache(ctx context.Context, store adt11.Store, cache migra
 		if err != nil {
 			return cid.Undef, xerrors.Errorf("failed to get previous outRoot from cache: %w", err)
 		}
+
+		okSectorIndexOut, prevSectorIndexRoot, err := cache.Read(migration.MinerPrevSectorDealIndexKey(inRoot))
```

Comment:
I think you want to look up `prevInRoot` here, not `inRoot`. We won't have a cached result for `inRoot` but we _should_ have one for `prevInRoot`.

Reply:
(By looking up `inRoot` I think you'll wind up doing the migration from scratch every time)

Reply:
changed here #a008d024f687a6fc24fc0ebac27b4961037343a5